src/vs/workbench/parts/extensions/electron-browser/extensionsViews.ts

```diff
@@ -320,7 +320,9 @@ export class ExtensionsListView extends CollapsibleView {
 				}
 
 				return this.extensionsWorkbenchService.queryGallery(assign(options, { names, pageSize: names.length }))
-					.then(pager => new PagedModel(pager));
+					.then(pager => this.addRecommendationInfo(new PagedModel(pager || []), {
+						source: 'fileType'
+					}));
```

Comment:
Not sure if it is good idea to add data to view model [IExtension](https://github.com/microsoft/vscode/blob/master/src/vs/workbench/parts/extensions/common/extensions.ts#L26). I think the real fact is [IGalleryExtension](https://github.com/microsoft/vscode/blob/master/src/vs/platform/extensionManagement/common/extensionManagement.ts#L130), the underlying mode that is passed around. This is what user is querying for installing. 

FYI I had a similar request to log search criteria and position of an extension when it is opened and installed. I captured this data in gallery extension and all telemetry logging will have this info. Please see [commit](https://github.com/Microsoft/vscode/commit/f117237ec50494e8c3a5bc67ab13c27bdf8a5dc3)

Reply:
Updating telemetry data on the paged model feels very wrong, now that I have seen how we use paged model later on in the viewlet

I like the idea of parsing search text in one place instead of 2, but I don't want to take that refactoring for 1.15 Will consider it for next milestone.


Going back to query options idea, for now I think this should be good:
- Add `source` property to IQueryOptions
- Pass the source to `toExtensions` which will add it to `telemetryData`

Reply:
Thanks. LGTM.